### PR TITLE
Resolving delete entity not working on empty primary keys

### LINKED ISSUE #36690: [BUG] Error deleting Table Entity with empty rowkey
**Bug Description:**
We are using azure-data-tables-12.3.14 library but getting issue in deleteting table entity with empty rowkey. This was allowed operation in older 8.6.6 library.
As per the document https://learn.microsoft.com/en-us/rest/api/storageservices/Designing-a-Scalable-Partitioning-Strategy-for-Azure-Table-Storage, empty stiring is permited for rowkey and partitionkey.

**Exception:**
2023-09-11 11:46:47.212 ERROR [           main]  com.azure.data.tables.TableClient        [MsgID=] : 'partitionKey' and 'rowKey' cannot be null 

**To Reproduce**
1. Add row in table storage with empty rowkey.
2. try to delete is using the method 
`public Response<Void> deleteEntityWithResponse(TableEntity entity, boolean ifUnchanged, Duration timeout, Context context)` avilable in the library.
3. The exception provided in the Exception section of this bug will be reproduced

***Code Snippet***
    private Response<Void> deleteEntityWithResponse(String partitionKey, String rowKey, String eTag, boolean ifUnchanged,
                                            Duration timeout, Context context) {
        Context contextValue = TableUtils.setContext(context, true);
        OptionalLong timeoutInMillis = TableUtils.setTimeout(timeout);

        String finalETag = ifUnchanged ? eTag : "*";

        if (isNullOrEmpty(partitionKey) || isNullOrEmpty(rowKey)) {
            throw logger.logExceptionAsError(new IllegalArgumentException("'partitionKey' and 'rowKey' cannot be null"));
        }

        Callable<Response<Void>> callable = () -> tablesImplementation.getTables().deleteEntityWithResponse(
            tableName, TableUtils.escapeSingleQuotes(partitionKey), TableUtils.escapeSingleQuotes(rowKey), finalETag, null,
            null, null, contextValue);

        try {
            return timeoutInMillis.isPresent()
                ? THREAD_POOL.submit(callable).get(timeoutInMillis.getAsLong(), TimeUnit.MILLISECONDS)
                : callable.call();
        } catch (Exception ex) {
            Throwable except = mapThrowableToTableServiceException(ex);
            return swallow404Exception(except);
        }
    }
    
**Expected behavior**
The exeption should not be thrown while deleting a row from the azure table storage with empty partitionKey and rowKey
row.
image

**Setup:**
 - OS: [Windows 11]
 - IDE: [Spring Toool Suite]
 - Library/Libraries: [com.azure:azure-data-tables:12.3.14 (groupId:artifactId:version)]
 - Java version: [17] 
 - App Server/Environment: [Tomcatl]
 - Frameworks: [Spring Boot] 

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

### PR DESCRIPTION
**Pull request by jairmyree**
This PR addresses the logic issue that prevented delete entities from being used on entities with an empty primary key.
It introduces new test for the deletion of entities with primary keys.

Closes #36690

### CODE REVIEW

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java [hunk 1]
```diff
@@ -840,8 +840,8 @@ Mono<Response<Void>> deleteEntityWithResponse(String partitionKey, String rowKey
         context = TableUtils.setContext(context);
         eTag = ifUnchanged ? eTag : "*";
 
-        if (isNullOrEmpty(partitionKey) || isNullOrEmpty(rowKey)) {
-            return monoError(logger, new IllegalArgumentException("'partitionKey' and 'rowKey' cannot be null."));
+        if (partitionKey == null || rowKey == null) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'partitionKey' and 'rowKey' cannot be null"));
```
Comment (srnagar) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java:844]: Add `@throws IllegalArgumentException` to the javadocs in both sync and async clients.

**Review by srnagar — APPROVED** [review]

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java (hunk 1 above)
Comment (mssfang) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java:844]: NIT: is `logger` should be a static `LOGGER`

**Review by mssfang — APPROVED** [review]

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java (hunk 1 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java:844]: Any particular reason we're moving away from using `monoError` like in the rest of the operations in the client?

Comment (jairmyree) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java:844]: That's not supposed to be there, it's supposed to be monoError

Comment (jairmyree) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java]: @mssfang What's the benefit of using a static LOGGER over a logger? It's always been non-static so I haven't questioned it. We can look into that in another PR.

Comment (mssfang) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java]: My bad. wrong suggestion. If the class is a static class then we accept having a shared logger instance

**Comment by maheshnaptemicrosoft** [comment]
Hi @jairmyree,

I noticed that this bug is closed. However, there is a similar issue with the following method in TableClient:
`@ServiceMethod(returns = ReturnType.SINGLE)
public Response<TableEntity> getEntityWithResponse(String partitionKey, String rowKey, List<String> select,

                                                       Duration timeout, Context context) {
                                                       ...
}`

**Pull request merged**